iothub_client/inc/internal/blob.h

```diff
@@ -57,10 +57,11 @@ MU_DEFINE_ENUM_WITHOUT_INVALID(BLOB_RESULT, BLOB_RESULT_VALUES)
 * @param  certificates      A null terminated string containing CA certificates to be used
 * @param    proxyOptions    A structure that contains optional web proxy information
 * @param  networkInterface    An optional null terminated string containing the network interface
+* @param  timeout           An optional timeout value for sending and receiving messages
 *
 * @return    A @c BLOB_RESULT. BLOB_OK means the blob has been uploaded successfully. Any other value indicates an error
 */
-MOCKABLE_FUNCTION(, BLOB_RESULT, Blob_UploadMultipleBlocksFromSasUri, const char*, SASURI, IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_CALLBACK_EX, getDataCallbackEx, void*, context, unsigned int*, httpStatus, BUFFER_HANDLE, httpResponse, const char*, certificates, HTTP_PROXY_OPTIONS*, proxyOptions, const char*, networkInterface)
+MOCKABLE_FUNCTION(, BLOB_RESULT, Blob_UploadMultipleBlocksFromSasUri, const char*, SASURI, IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_CALLBACK_EX, getDataCallbackEx, void*, context, unsigned int*, httpStatus, BUFFER_HANDLE, httpResponse, const char*, certificates, HTTP_PROXY_OPTIONS*, proxyOptions, const char*, networkInterface, const long*, timeout)
```

Comment:
Suggestion: rename `timeout` to `timeoutÌnMilliseconds`

Reply:
Why not pass const long instead of the pointer, and check for 0 within Blob_UploadMultipleBlocksFromSasUri ?

Reply:
I was thinking the parameter had to be nullable because of the other parameters (and since it's optional), but it doesn't look like it does since this is an internal function - I'll make this change

Reply:
I couldn't pass it as a const or the unit tests got mad (`Could not copy type, type const size_t not registered.`) - if there's a way around this, I'm happy to change it back to a const though!

Reply:
I think you might be able to get around it with a registered alias type:

https://github.com/Azure/azure-c-shared-utility/blob/718ec00a1df4917f1ee8aa528bdfa37e56bd8e92/tests/tlsio_options_ut/tlsio_options_ut.c#L95

Reply:
That worked, thanks!